src/Aggregation/DateHistogram.php

```diff
@@ -11,6 +11,59 @@ class DateHistogram extends Histogram
 {
     public const DEFAULT_TIMEZONE_VALUE = 'UTC';
 
+    private const CALENDAR_INTERVAL_VALUES = [
+        '1m',
+        'minute',
+        '1h',
+        'hour',
+        '1d',
+        'day',
+        '1w',
+        'week',
+        '1M',
+        'month',
+        '1q',
+        'quarter',
+        '1y',
+        'year',
+    ];
+
+    /**
+     * Set the interval by which documents will be bucketed.
+     *
+     * @param int|string $interval
+     *
+     * @return $this
+     */
+    public function setInterval($interval): Histogram
+    {
+        if (\in_array($interval, self::CALENDAR_INTERVAL_VALUES, true)) {
+            return $this->setCalendarInterval($interval);
+        }
+
+        return $this->setFixedInterval($interval);
+    }
+
+    /**
+     * @param int|string $interval
+     *
+     * @return $this
+     */
+    public function setCalendarInterval($interval): self
```

Comment:
```suggestion
    /**
     * @return $this
     */
    public function setCalendarInterval(string $interval): self
```
What about creating these methods with the right type? `setInterval` allows `int|string`, but in [the ES documentation](https://www.elastic.co/guide/en/elasticsearch/reference/7.14/search-aggregations-bucket-datehistogram-aggregation.html#calendar_and_fixed_intervals) seems like it only accepts `string`.

It should be fine since we don't have `strict_types` so the value from `setInterval` will be casted to `string`.

Reply:
The problem now is that `setInterval()` accepts an `int|string`, while the two methods it calls, they expect a `string`.

I guess we can not force `setInterval(string $internal)` as signature, as it will not be compatible with the parent class